### DISCUSSION TITLE
How do can one pythonically hide a Field's metadata from openapi.json?

### DISCUSSION THREAD

**Discussion by bauerem** [fastapi/fastapi#11776] [Questions] [answered]
### First Check

- [x] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from pydantic import BaseModel, Field

class ExampleData(BaseModel):
    text1 : str = Field('Hello World!', metadata={"postprocess": True})
    text2 : str = Field('Hello World!', metadata={"postprocess": False})

app = FastAPI(
)

@app.get('/example')
def get_data() -> ExampleData:
    return ExampleData()
```

### Description

How can I add metadata to Fields without the metadata being accessible from outside the service?
The metadata appears in /openapi.json but it shouldn't be visible to users.
I would rather not use the Annotated class for this functionality as I think it is a lot more clunky and unintuitive than adding the metadata to the Field. Is something like this possible?

Why: I want to add metadata to my Pydantic schema because I want to process the data of some fields before sending it to another service which shares the same data schema.

### Operating System

Linux, macOS

### FastAPI Version

0.110.0

### Pydantic Version

2.7.2

### Python Version

3.12.3

**Comment by YuriiMotov** [comment]
Passing extra parameters to `Field` is deprecated:
>     extra: (Deprecated) Extra fields that will be included in the JSON schema.
> 
>         !!! warning Deprecated
>             The extra kwargs is deprecated. Use json_schema_extra instead.

**Reply by bauerem** [reply]
Thank you for your reply.
In the FastAPI docs it says about json_schema_extra:
"That extra info will be added as-is to the output JSON Schema for that model, and it will be used in the API docs."
That's what I am trying to prevent.
How would you recommend adding data associated to Fields while hiding it from the docs?

**Reply by YuriiMotov** [reply, marked as answer]
I would use `Annotated`
